db/version_set.cc

```diff
@@ -1009,9 +1009,16 @@ class LevelIterator final : public InternalIterator {
     if (range_tombstone_iter_ptr_) {
       *range_tombstone_iter_ptr_ = &range_tombstone_iter_;
     }
+
+    prepared_iters.reserve(file_index_ + 1);
   }
 
-  ~LevelIterator() override { delete file_iter_.Set(nullptr); }
+  ~LevelIterator() override {
+    delete file_iter_.Set(nullptr);
+    for (auto v : prepared_iters) {
+      delete v;
```

Comment:
If SetFileIterator(prepared_iters[..]) was called, would we double delete on the same pointer here?